src/Foundatio.Extensions.Hosting/Jobs/JobHostExtensions.cs

```diff
@@ -33,18 +36,19 @@ public static IServiceCollection AddJob(this IServiceCollection services, Func<I
         {
             return services.AddTransient<IHostedService>(s =>
             {
-                jobOptions.JobFactory = () => jobFactory(s);
+                jobOptions.JobFactory = jobFactory;
 
                 return new HostedJobService(s, jobOptions, s.GetService<ILoggerFactory>());
             });
         }
-        else
-        {
-            if (!services.Any(s => s.ServiceType == typeof(IHostedService) && s.ImplementationType == typeof(ScheduledJobService)))
-                services.AddTransient<IHostedService, ScheduledJobService>();
 
-            return services.AddTransient(s => new ScheduledJobRegistration(() => jobFactory(s), jobOptions.CronSchedule));
-        }
+        if (!services.Any(s => s.ServiceType == typeof(IHostedService) && s.ImplementationType == typeof(ScheduledJobService)))
+            services.AddTransient<IHostedService, ScheduledJobService>();
+
+        if (!services.Any(s => s.ServiceType == typeof(IScheduledJobManager) && s.ImplementationType == typeof(ScheduledJobManager)))
+            services.AddSingleton<ScheduledJobManager>();
+
+        return services.AddTransient(s => new ScheduledJobRegistration(jobOptions.CronSchedule, jobOptions.Name ?? Guid.NewGuid().ToString(), _ => jobFactory(s)));
```

Comment:
Kind of feels like job name should be required. If you just use a guid, it really breaks the manager.

Reply:
It should already be always populated, but you don't need a name on every job and lots of jobs you would never update.